### PR TITLE
add metalinter to CI and include fixes

### PR DESCRIPTION
**Pull request by zramsay**
- replaces #643

### CODE REVIEW

File: benchmarks/map_test.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package benchmarks
+package benchmarks // nolint: goimports
```
Comment (ebuchman) [inline benchmarks/map_test.go]: As far as I know, there's no case ever wherein we'd want to use nolint for goimports. Please correct me if I'm wrong

**Comment by ebuchman** [comment]
Also, see https://github.com/tendermint/tendermint/issues/715

We should add that to the coding repo and reflect it here

File: benchmarks/map_test.go (hunk 1 above)
Comment (zramsay) [reply benchmarks/map_test.go]: thought i'd removed them all

**Comment by ebuchman** [comment]
Hey @zramsay can we get one more rebase here plx

File: circle.yml [hunk 2]
```diff
@@ -24,6 +24,7 @@ dependencies:
 
 test:
   override:
+    - cd "$PROJECT_PATH" && make get_vendor_deps && make metalinter_test
```
Comment (ebuchman) [inline circle.yml]: can we do the metalinter test without vendor deps? we're going to download the vendor deps later ... no need to do it twice

**Comment by ebuchman** [comment]
Think this is good to merge. Just waiting for CI :)

File: test/run_test.sh [hunk 3]
```diff
@@ -6,6 +6,10 @@ pwd
 BRANCH=$(git rev-parse --abbrev-ref HEAD)
 echo "Current branch: $BRANCH"
 
+# run the linter
+# TODO: drop the `_test` once we're ballin' enough
```
Comment (zramsay) [inline test/run_test.sh]: :1st_place_medal:

**Pull request merged**